pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/BaseSegmentPushJobRunner.java

```diff
@@ -50,12 +50,19 @@ public abstract class BaseSegmentPushJobRunner implements IngestionJobRunner {
 
   /**
    * Initialize BaseSegmentPushJobRunner with SegmentGenerationJobSpec
-   * Checks for required parameters in the spec and enablement of consistent data push.
    */
   @Override
   public void init(SegmentGenerationJobSpec spec) {
+    init(spec, true);
+  }
+
+  /**
+   * Initialize BaseSegmentPushJobRunner with SegmentGenerationJobSpec
+   * Checks for required parameters in the spec and enablement of consistent data push.
+   */
+  public void init(SegmentGenerationJobSpec spec, boolean checkPushJobSpec) {
```

Comment:
instead of adding `checkPushJobSpec` boolean flag, let's keep the `BaseSegmentPushJobRunner` as it is and override init() function within `SegmentTarPushJobRunner` with the old logic. Let's double check about `tableSpec, tableConfigURI` as well.

Reply:
Done. 
1. Kept `tableSpec` as required as a NPE will happen downstream if not supplied.
2. Made fix to have `tableSpec.tableConfigURI` as optional.